specification/maps/data-plane/Common/preview/1.0/common.json

```diff
@@ -1,712 +1,712 @@
-      "allOf": [
-        {
-          "$ref": "#/definitions/GeoJsonObject"
-        }
-      ]
-    },
-    "GeoJsonPoint": {
-      "description": "A valid `GeoJSON Point` geometry type. Please refer to [RFC 7946](https://tools.ietf.org/html/rfc7946#section-3.1.2) for details.",
-      "x-ms-discriminator-value": "Point",
-      "type": "object",
-      "allOf": [
-        {
-          "$ref": "#/definitions/GeoJsonGeometry"
-        },
-        {
-          "$ref": "#/definitions/GeoJsonPointData"
-        }
-      ]
-    },
-    "GeoJsonPointData": {
-      "description": "Data contained by a `GeoJson Point`.",
-      "type": "object",
-      "properties": {
-        "coordinates": {
+      },
+      "GeoJsonLinearRing": {
+        "description": "Though a linear ring is not explicitly represented as a GeoJSON geometry type, it helps in defining the polygon as an array of linear rings. Please refer to [RFC 7946](https://tools.ietf.org/html/rfc7946#section-3.1.6) for details.",
+        "type": "array",
+        "items": {
           "$ref": "#/definitions/GeoJsonPosition"
         }
       },
-      "required": [
-        "coordinates"
-      ]
-    },
-    "GeoJsonMultiPoint": {
-      "description": "A valid `GeoJSON MultiPoint` geometry type. Please refer to [RFC 7946](https://tools.ietf.org/html/rfc7946#section-3.1.3) for details.",
-      "x-ms-discriminator-value": "MultiPoint",
-      "type": "object",
-      "allOf": [
-        {
-          "$ref": "#/definitions/GeoJsonGeometry"
-        },
-        {
-          "$ref": "#/definitions/GeoJsonMultiPointData"
-        }
-      ]
-    },
-    "GeoJsonMultiPointData": {
-      "type": "object",
-      "description": "Data contained by a `GeoJson MultiPoint`.",
-      "properties": {
-        "coordinates": {
-          "description": "Coordinates for the `GeoJson MultiPoint` geometry.",
-          "type": "array",
-          "items": {
-            "$ref": "#/definitions/GeoJsonPosition"
+      "GeoJsonObject": {
+        "description": "A valid `GeoJSON` object. Please refer to [RFC 7946](https://tools.ietf.org/html/rfc7946#section-3) for details.",
+        "type": "object",
+        "discriminator": "type",
+        "required": [
+          "type"
+        ],
+        "properties": {
+          "type": {
+            "description": "Specifies the `GeoJSON` type. Must be one of the nine valid GeoJSON object types - Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon, GeometryCollection, Feature and FeatureCollection.",
+            "type": "string",
+            "enum": [
+              "Point",
+              "MultiPoint",
+              "LineString",
+              "MultiLineString",
+              "Polygon",
+              "MultiPolygon",
+              "GeometryCollection",
+              "Feature",
+              "FeatureCollection"
+            ],
+            "x-ms-enum": {
+              "name": "GeoJsonObjectType",
+              "modelAsString": true,
```

Comment:
Since this is a closed enum, `modelAsString` may not be appropriate here.

Reply:
Made it a closed enum in [this commit](https://github.com/Azure/azure-rest-api-specs/pull/16030/commits/eadc4f90122ac15d7837fafa9a9a65e51dd20b46)